### PR TITLE
Bootstrap multi-environments

### LINKED ISSUE #3: Add multiple environments (as config files) and store multiple state files
Instead of needing to create a private fork per bootstrap environment (i.e. for a dev bootstrapping env and a prod one), we should be able to add multiple config.yaml files, one per environment (i.e. config.dev.yaml, config.prod.yaml), and be able to select the one we'd like to deploy via an env var, switch etc. 

In turn this should create and reference a separate state file within the repo for the environment to allow check-in.

### PR DESCRIPTION
**Pull request by jjgriff93**
Resolves #3 

Adds ability to deploy more than one bootstrap environment from the same repo

- Looks for config.${environment}.yaml files for environment-specific settings
- Store separate local state per environment

### CODE REVIEW

File: config.sample.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 ---
-suffix: samplebootstrap
+bootstrap_id: samplebootstrap
```
Comment (t-young31) [inline config.sample.yaml:15]: consistency ftw ❤️

File: deployment/locals.tf [hunk 2]
```diff
@@ -13,7 +13,8 @@
 #  limitations under the License.
 
 locals {
-  suffix_truncated  = replace(replace(var.suffix, "-", ""), "_", "")
+  suffix            = "${var.bootstrap_id}-${var.environment}"
+  suffix_truncated  = replace(replace(local.suffix, "-", ""), "_", "")
```
Comment (t-young31) [inline deployment/locals.tf:17]:
```suggestion
  suffix_truncated  = substr(replace(replace(local.suffix, "-", ""), "_", ""), 0, 17)
```

File: config.sample.yaml (hunk 1 above)
Comment (jjgriff93) [reply config.sample.yaml:15]: Exactly :)

File: deployment/locals.tf (hunk 2 above)
Comment (jjgriff93) [reply deployment/locals.tf:17]: Nice!

File: deployment/terraform.tf
[the remark this answers is not in the document]
Comment (jjgriff93) [reply deployment/terraform.tf:1]: Right you are sir

**Pull request merged**